### ISSUE TITLE
Postgres for Clair is not deployed if existing DB is used for Quay

### ISSUE THREAD

**Issue by kurtismullins** [quay/quay-operator#111]
While testing 0.0.8, it was discovered that using a CR to manually specify an existing Postgres for Quay stops the operator from deploying a Postgres database for Clair.

If this is the intended behaviour, it may be a matter of updating the documentation.

**Expected Behavior:**
- Postgres is deployed for Clair

**Actual Behavior:**
- No postgres is deployed.

**Example CR**
```
spec:
  quay:
    database:
      credentialsSecretName: postgresql-account
      server: 172.30.98.48:5432
  clair:
    enabled: true
    imagePullSecretName: quay.io
    database:
      volumeSize: 3Gi
```

**Comment by sabre1041** [comment]
@kurtismullins would you be able to retest and confirm whether this is still an issue?

**Comment by gripenk** [comment]
i am testing 1.0.1 on okd3.11, the clair is still not deployed.

oc get all
NAME                                                         READY     STATUS    RESTARTS   AGE
pod/example-quayecosystem-quay-7f8694d68f-ndkv4              1/1       Running   0          6m
pod/example-quayecosystem-quay-postgresql-544c5ccbcf-9jzl8   1/1       Running   0          1h
pod/example-quayecosystem-quay-repomirror-75bc4b4bf9-k48dr   1/1       Running   0          52m
pod/example-quayecosystem-redis-6d8d4984df-rqqll             1/1       Running   0          1h
pod/quay-operator-7d766f4cbf-2p97w                           1/1       Running   0          57m

NAME                                            TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)             AGE
service/example-quayecosystem-quay              ClusterIP   172.30.151.85   <none>        443/TCP             1h
service/example-quayecosystem-quay-config       ClusterIP   172.30.215.46   <none>        443/TCP             1h
service/example-quayecosystem-quay-postgresql   ClusterIP   172.30.6.16     <none>        5432/TCP            1h
service/example-quayecosystem-redis             ClusterIP   172.30.8.176    <none>        6379/TCP            1h
service/quay-operator-metrics                   ClusterIP   172.30.18.80    <none>        8383/TCP,8686/TCP   1h

NAME                                                    DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/example-quayecosystem-quay              1         1         1            1           1h
deployment.apps/example-quayecosystem-quay-postgresql   1         1         1            1           1h
deployment.apps/example-quayecosystem-quay-repomirror   1         1         1            1           52m
deployment.apps/example-quayecosystem-redis             1         1         1            1           1h
deployment.apps/quay-operator                           1         1         1            1           1h

NAME                                                               DESIRED   CURRENT   READY     AGE
replicaset.apps/example-quayecosystem-quay-55fc7dcdd7              0         0         0         1h
replicaset.apps/example-quayecosystem-quay-6d95dcb9c               0         0         0         1h
replicaset.apps/example-quayecosystem-quay-7f8694d68f              1         1         1         6m
replicaset.apps/example-quayecosystem-quay-postgresql-544c5ccbcf   1         1         1         1h
replicaset.apps/example-quayecosystem-quay-postgresql-79cfc54855   0         0         0         1h
replicaset.apps/example-quayecosystem-quay-repomirror-75bc4b4bf9   1         1         1         52m
replicaset.apps/example-quayecosystem-redis-597d4575f5             0         0         0         1h
replicaset.apps/example-quayecosystem-redis-69689cbfb7             0         0         0         1h
replicaset.apps/example-quayecosystem-redis-6d8d4984df             1         1         1         1h
replicaset.apps/quay-operator-7d766f4cbf                           1         1         1         57m

NAME                                                         HOST/PORT                                                                         PATH      SERVICES                            PORT      TERMINATION            WILDCARD
route.route.openshift.io/example-quayecosystem-quay          example-quayecosystem-quay-quay-enterprise                    example-quayecosystem-quay          8443      passthrough/Redirect   None
route.route.openshift.io/example-quayecosystem-quay-config   example-quayecosystem-quay-config-quay-enterprise             example-quayecosystem-quay-config   8443      passthrough/Redirect   None

**Comment by sabre1041** [comment]
@gripenk can you share your `QuayEcosystem` CR?

**Comment by gripenk** [comment]
I have updated some image source to registry.redhat.io

apiVersion: redhatcop.redhat.io/v1alpha1
kind: QuayEcosystem
metadata:
  creationTimestamp: '2020-01-11T17:38:02Z'
  generation: 13
  name: example-quayecosystem
  namespace: quay-enterprise
  resourceVersion: '55406310'
  selfLink: >-
    /apis/redhatcop.redhat.io/v1alpha1/namespaces/quay-enterprise/quayecosystems/example-quayecosystem
  uid: 1da7b6d9-3499-11ea-9c85-0050568364d3
spec:
  clair:
    database:
      deploymentStrategy: RollingUpdate
      image: registry.redhat.io/rhscl/postgresql-96-rhel7
      livenessProbe:
        exec:
          command:
            - /usr/libexec/check-container
            - '--live'
        failureThreshold: 3
        initialDelaySeconds: 120
        timeoutSeconds: 10
      readinessProbe:
        exec:
          command:
            - /usr/libexec/check-container
        failureThreshold: 3
        initialDelaySeconds: 10
        timeoutSeconds: 1
      resources: {}
    deploymentStrategy: RollingUpdate
    enabled: true
    image: 'quay.io/redhat/clair-jwt:v3.2.0'
    imagePullSecretName: redhat-pull-secret
    livenessProbe:
      failureThreshold: 3
      httpGet:
        path: /health
        port: 6061
        scheme: HTTP
      initialDelaySeconds: 30
      timeoutSeconds: 5
    readinessProbe:
      failureThreshold: 3
      httpGet:
        path: /health
        port: 6061
        scheme: HTTP
      initialDelaySeconds: 10
      timeoutSeconds: 5
    resources: {}
    updateInterval: 60m
  quay:
    configResources: {}
    database:
      deploymentStrategy: RollingUpdate
      image: registry.redhat.io/rhscl/postgresql-96-rhel7
      livenessProbe:
        exec:
          command:
            - /usr/libexec/check-container
            - '--live'
        failureThreshold: 3
        initialDelaySeconds: 120
        timeoutSeconds: 10
      readinessProbe:
        exec:
          command:
            - /usr/libexec/check-container
        failureThreshold: 3
        initialDelaySeconds: 10
        timeoutSeconds: 1
      resources: {}
      volumeSize: 10Gi
    deploymentStrategy: RollingUpdate
    enableRepoMirroring: true
    image: 'quay.io/redhat/quay:v3.2.0'
    imagePullSecretName: redhat-pull-secret
    livenessProbe:
      failureThreshold: 3
      httpGet:
        path: /health/instance
        port: 8443
        scheme: HTTPS
      initialDelaySeconds: 120
      timeoutSeconds: 5
    readinessProbe:
      failureThreshold: 3
      httpGet:
        path: /health/instance
        port: 8443
        scheme: HTTPS
      initialDelaySeconds: 5
    registryBackends:
      - local:
          storagePath: /datastorage/registry
        name: default
    registryStorage:
      persistentVolumeAccessModes:
        - ReadWriteMany
      persistentVolumeSize: 10Gi
      persistentVolumeStorageClassName: managed-nfs-storage-1
    repoMirrorResources: {}
    resources: {}
  redis:
    deploymentStrategy: RollingUpdate
    image: registry.redhat.io/rhscl/redis-5-rhel7
    readinessProbe:
      failureThreshold: 3
      initialDelaySeconds: 30
      tcpSocket:
        port: 6379
    resources: {}
status:
  conditions:
    - lastTransitionTime: '2020-01-11T17:38:02Z'
      lastUpdateTime: '2020-01-11T19:18:50Z'
      message: Configuration Updated Successfully
      status: 'True'
      type: UpdateDefaultConfigurationSuccess
    - lastTransitionTime: '2020-01-11T17:44:46Z'
      lastUpdateTime: '2020-01-11T17:44:46Z'
      message: 'Failed to Exec into Postgresql Pod: '
      reason: ProcessingError
      status: 'False'
      type: QuayEcosystemProvisioningFailure
    - lastTransitionTime: '2020-01-11T17:48:31Z'
      lastUpdateTime: '2020-01-11T17:48:31Z'
      message: Setup Completed Successfully
      status: 'True'
      type: QuaySetupSuccess
    - lastTransitionTime: '2020-01-11T18:12:24Z'
      lastUpdateTime: '2020-01-12T10:07:08Z'
      message: >-
        Get https://example-quayecosystem-quay-config/api/v1/superuser/keys:
        dial tcp 172.30.215.46:443: connect: no route to host
      reason: ProcessingError
      status: 'False'
      type: QuayEcosystemSecurityScannerConfigurationFailure
  hostname: example-quayecosystem-quay-quay-enterprise.cloudapps-itscloud-dev.hku.hk
  setupComplete: true

**Comment by kurtismullins** [comment]
@gripenk This appears to be resolved for me in 1.0.1. Would you mind to verify that you're still experiencing the same issue? Thank you!

**Comment by ricardomaraschini** [comment]
QuayEcosystem has been deprecated. This operator now uses QuayRegistry instances instead. I am closing this one, feel free to reopen if necessary.
